sdks/python/apache_beam/runners/direct/transform_evaluator.py

```diff
@@ -212,6 +218,19 @@ class _TestStreamRootBundleProvider(RootBundleProvider):
   """
   def get_root_bundles(self):
     test_stream = self._applied_ptransform.transform
+    if self._evaluation_context._test_stream_event_stub:
+      stub = self._evaluation_context._test_stream_event_stub
+
+      if list(test_stream.output_tags) == [None]:
+        event_request = beam_runner_api_pb2.EventsRequest()
+      else:
+        event_request = beam_runner_api_pb2.EventsRequest(
+            keys=list(test_stream.output_tags))
```

Comment:
Why does the set of keys even need to be sent here?

Reply:
To know which PCollections to read from. Added a comment.